File: memory-api/src/main/java/com/memory/service/calendar/CalendarEventServiceUtils.java
package com.memory.service.calendar;

import com.memory.domain.member.Member;
import com.memory.domain.relationship.Relationship;
import com.memory.exception.customException.ValidationException;

public class CalendarEventServiceUtils {

    public static void validateRelationshipMember(Member member, Relationship relationship) {
        if (!relationship.getMember().getId().equals(member.getId()) &&
            !relationship.getRelatedMember().getId().equals(member.getId())) {
            throw new ValidationException("해당 관계에 속한 회원이 아닙니다.");
        }
    }

}

File: memory-api/src/main/kotlin/com/memory/service/calendar/CalendarEventServiceUtils.kt
package com.memory.service.calendar

import com.memory.domain.member.Member
import com.memory.domain.relationship.Relationship
import com.memory.exception.customException.ValidationException

object CalendarEventServiceUtils {
    fun validateRelationshipMember(member: Member?, relationship: Relationship?) {
        if (relationship?.member?.id != member?.id && relationship?.relatedMember?.id != member?.id) {
            throw ValidationException("해당 관계에 속한 회원이 아닙니다.")
        }
    }
}
